### ISSUE TITLE
Correcting a typo make it perform worse

### ISSUE THREAD

**Issue by tommyvct** [chuyuanli/Seq2Seq-DDP#4]
In `transition_predict.py`, the prompt for the task is `"dicourse parsing: "`, while in `train.py` the prompt is `"discourse parsing: "`, without typo. 

With the same t0-3b model, finetuned on the molweni dataset, learning rate of `5e-5`, 5 epochs, batch size of 4, steps of 2000, inferencing with and without the typo gives different result: 

| \ | With typo | without typo |
|---------|----------|----------|
| focus  | `[link+rel] recall: 59.15, precision: 58.89, f1: 59.02  [linkonly] recall: 82.91, precision: 82.63, f1: 82.77`   | `[link+rel] recall: 57.47, precision: 57.22, f1: 57.35 [linkonly] recall: 81.64, precision: 81.36, f1: 81.5`  |
| nat2  | `[link+rel] recall: 59.51, precision: 59.26, f1: 59.38 [linkonly] recall: 83.54, precision: 83.26, f1: 83.4`   | `[link+rel] recall: 57.65, precision: 57.4, f1: 57.53 [linkonly] recall: 81.23, precision: 80.95, f1: 81.09`   |

Note that all of the above results are inferred from the same model fine-tuned on molweni.

It looks like the typo-free version of inference performs worse than the one with typo. Why?

**Comment by chuyuanli** [comment]
This is interesting, and thank you for pointing out the typo in the `transition_predict.py` script. 
The instruction prefix should be the same in training and prediction (whether both with typo or typo-free). 
The typo might matter in terms of tokenization change, hence decoder behavior. Another possible reason could also be normal run-to-run variance reading the relatively small performance gap. 

I think a rather comprehensive way to test: 
A: train with the typo in the prompt, hold the checkpoint and everything else fixed, vary the predict prompt, and repeat across several seeds.
B: train without the typo in the prompt, hold the checkpoint and everything else fixed, vary the predict prompt, and repeat across several seeds.

Of course, an even more complete way is to repeat A B tests each with several train runs, so that we are considering the training variance not just inference variance.

Hope this helps.
